pkg/varlinkapi/images.go

```diff
@@ -1,75 +1,285 @@
+	cliutil "github.com/projectatomic/libpod/cmd/podman/util"
+	rt "github.com/projectatomic/libpod/cmd/podman/util"
+	"github.com/projectatomic/libpod/libpod/image"
+	sysreg "github.com/projectatomic/libpod/pkg/registries"
 )
 
-// ListImages ...
+// ListImages lists all the images in the store
+// It requires no inputs.
 func (i *LibpodAPI) ListImages(call ioprojectatomicpodman.VarlinkCall) error {
-	return call.ReplyMethodNotImplemented("ListImages")
+	runtime, err := rt.GetRuntime(i.Cli)
+	if err != nil {
+		return call.ReplyRuntimeError(err.Error())
+	}
+	images, err := runtime.ImageRuntime().GetImages()
+	if err != nil {
+		return call.ReplyErrorOccurred(fmt.Sprintf("unable to get list of images %q", err))
+	}
+	var imageList []ioprojectatomicpodman.ImageInList
+	for _, image := range images {
+		//size, _:= image.Size(getContext())
+		labels, _ := image.Labels(getContext())
+		containers, _ := image.Containers()
+
+		i := ioprojectatomicpodman.ImageInList{
+			Id:          image.ID(),
+			ParentId:    image.Parent,
+			RepoTags:    image.Names(),
+			RepoDigests: image.RepoDigests(),
+			Created:     image.Created().String(),
+			//Size: size,
+			VirtualSize: image.VirtualSize,
+			Containers:  int64(len(containers)),
+			Labels:      labels,
+		}
+		imageList = append(imageList, i)
+	}
+	return call.ReplyListImages(imageList)
 }
 
 // BuildImage ...
+// TODO Waiting for buildah to be vendored into libpod to do this only one
 func (i *LibpodAPI) BuildImage(call ioprojectatomicpodman.VarlinkCall) error {
 	return call.ReplyMethodNotImplemented("BuildImage")
 }
 
 // CreateImage ...
+// TODO With Pull being added, should we skip Create?
 func (i *LibpodAPI) CreateImage(call ioprojectatomicpodman.VarlinkCall) error {
 	return call.ReplyMethodNotImplemented("CreateImage")
 }
 
-// InspectImage ...
-func (i *LibpodAPI) InspectImage(call ioprojectatomicpodman.VarlinkCall) error {
-	return call.ReplyMethodNotImplemented("InspectImage")
+// InspectImage returns an image's inspect information as a string that can be serialized.
+// Requires an image ID or name
+func (i *LibpodAPI) InspectImage(call ioprojectatomicpodman.VarlinkCall, name string) error {
+	runtime, err := rt.GetRuntime(i.Cli)
+	if err != nil {
+		return call.ReplyRuntimeError(err.Error())
+	}
+	newImage, err := runtime.ImageRuntime().NewFromLocal(name)
+	if err != nil {
+		return call.ReplyImageNotFound(name)
+	}
+	inspectInfo, err := newImage.Inspect(getContext())
+	b, err := json.Marshal(inspectInfo)
+	if err != nil {
+		return call.ReplyErrorOccurred(fmt.Sprintf("unable to serialize"))
+	}
+	return call.ReplyInspectImage(string(b[:]))
 }
 
-// HistoryImage ...
-func (i *LibpodAPI) HistoryImage(call ioprojectatomicpodman.VarlinkCall) error {
-	return call.ReplyMethodNotImplemented("HistoryImage")
+// HistoryImage returns the history of the image's layers
+// Requires an image or name
+func (i *LibpodAPI) HistoryImage(call ioprojectatomicpodman.VarlinkCall, name string) error {
```

Comment:
Wait, this returns just `error`... Are the normal `call.ReplyHistoryImage()` responses considered errors? Ew.

Reply:
the return is defined in the generated file. it's a bit of a thing to get used to, but this is correct as is.